File: web3-libs/mod4/playground/aula3/getTransactionReceipt.js
// 1. Instale e importe a biblioteca Ethers.js
import { ethers } from "ethers";
import readline from "readline";

// 2. Configure o provider
const RPC_URL = "http://127.0.0.1:8545";
const provider = new ethers.JsonRpcProvider(RPC_URL);

// 3. Obtenha o endereço da conta via readline input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Digite o endereço da conta: ", (txHash) => {
  provider
    .getTransactionReceipt(txHash)
    .then((transaction) => {
      console.log(`Transação de ${txHash}`, transaction);
    })
    .catch((error) => {
      console.error("Erro ao buscar a transação:", error);
    });
  rl.close();
});

File: web3-libs/mod3/playground/aula3/get_transaction_receipt.py
# 1. Instale e importe a biblioteca web3.py
from pprint import pprint
from web3 import Web3

# 2. Configure o provider
RPC_URL = "http://127.0.0.1:8545"
web3 = Web3(Web3.HTTPProvider(RPC_URL))

# 3. Obtenha o hash da transação via input
tx_hash = input("Digite o hash da transação: ")

try:
    transaction_receipt = web3.eth.get_transaction_receipt(tx_hash)
    print("Logs da Transação:", len(transaction_receipt.logs))

    for log in transaction_receipt.logs:
        print("Log:")
        pprint(dict(log))
except Exception as error:
    print("Erro ao buscar a transação:", error)
